config/application.pantheon.php

```diff
@@ -0,0 +1,40 @@
+<?php
+/**
+ * Pantheon platform application settings.
+ *
+ * IMPORTANT NOTE:
+ * Do not modify this file. This file is maintained by Pantheon.
```

Comment:
Usually, Pantheon-managed files in a composer-managed site are brought in via a dependency, e.g. pantheon-systems/drupal-integrations for Drupal sites. For WordPress composer-managed sites, this pattern was not followed because there were not many files to manage this way. Is there any appetite to revisit this decision?

Reply:
Ultimately, I don't know that that would leave us in a better place than adding a file for Pantheon-maintained modifications as far as merge conflicts. Currently, we're making a one-line change in a part of `application.php` that is unlikely to change much, if they've changed it at all. If we added a dependency and moved our modifications there, we'd make changes to the `composer.json`, which users almost certainly _have_ made modifications to.

Also, environment-specific `application.php` files is a [recommended practice for Bedrock](https://github.com/roots/bedrock/blob/master/config/application.php#L143-L147). We're going against the grain a little bit by not using the `/environment/` directory, but since our changes apply to the platform and not a specific environment, this seemed to make the most sense. If people are familiar with Bedrock already, having a Pantheon-specific config file won't seem out of place.

I'm not broadly opposed to the idea, but this is also code that would otherwise have already existed in `wp-config.php` (or `wp-config-pantheon.php`) and I think it would be weirder to _not_ make it part of the core project.